mongo/options/findoptions.go

```diff
@@ -348,6 +352,8 @@ type FindOneOptions struct {
 
 	// If true, the cursor created by the operation will not timeout after a period of inactivity. The default value
 	// is false.
+	//
+	// Deprecated: This option is not valid for a findOne operation, as no cursor is actually created.
 	NoCursorTimeout *bool
 
 	// This option is for internal replication use only and should not be set.
```

Comment:
`OplogReplay` should be marked deprecated too, though for a different reason. It was deprecated in MongoDB 4.4 (and is ignored by 4.4+ servers per SERVER-36186). The `OplogReplay` on the regular `FindOptions` was deprecated, and I think this will have the same note.

A bit of a tangent, but this brought back memories of a discussion about whether it's right to deprecate options in the driver that are deprecated in later server releases. I recall the takeaway was that server options that are planned for removal in the server should be deprecated in drivers and emit a deprecation warning. They may be valid to use on older server versions, but users should have fair warning that the option will be removed in a major release of the driver.

Reply:
Sounds good; deprecated `OplogReplay` with a note mentioning 4.4. 

Yeah, the discussion in [36816](https://jira.mongodb.org/browse/SERVER-36186) seems to reflect your point; although, as of 2018, there was no plan to remove `OplogReplay`, as many customers still used it to tail the oplog.

Reply:
That's true, I don't see a plan for removal. I suppose in the new Version API world, the server has a better process for deprecating / removing options.

Reply:
It looks like we deprecated `OplogReplay` in `FindOptions` [here](https://github.com/mongodb/mongo-go-driver/blob/b2aec8d38f669560c732e6675ead1ec53c6b9ec1/mongo/options/findoptions.go#L74) during the 4.4 release cycle but didn't deprecate it in `FindOneOptions`. I'm OK making that change as part of this commit, but can we copy over the same deprecation message as the one in the link for consistency?

Reply:
Ah didn't see that it had been deprecated in `FindOptions` already; would've copied that message. Yep! Updated to use the previous message.